plugin/evm/gossiper_orders.go

```diff
@@ -66,19 +65,19 @@ func (n *pushGossiper) gossipSignedOrders() (int, error) {
 	n.lastOrdersGossiped = time.Now()
 	now := time.Now().Unix()
 	selectedOrders := []*hu.SignedOrder{}
-	for orderHash, order := range n.ordersToGossip {
+	for _, order := range n.ordersToGossip {
 		if len(selectedOrders) >= maxSignedOrdersGossipBatchSize {
 			break
 		}
 		if order.ExpireAt.Int64() < now {
 			n.stats.IncSignedOrdersGossipOrderExpired()
 			log.Warn("signed order expired before gossip", "order", order, "now", now)
-			delete(n.ordersToGossip, orderHash)
 			continue
 		}
 		selectedOrders = append(selectedOrders, order)
-		delete(n.ordersToGossip, orderHash)
 	}
+	// delete all selected orders from n.ordersToGossip
+	n.ordersToGossip = n.ordersToGossip[len(selectedOrders):]
```

Comment:
`len(selectedOrders)` doesn't account for orders that were not added because they were expired, we gotta skip those too.